mods/d2k/maps/atreides-01a/atreides01a.lua

```diff
@@ -0,0 +1,159 @@
+		Media.DisplayMessage(Messages[3], "Mentat")
+	end
+
+	if HasPower and player.Resources > player.ResourceCapacity * 0.8 and DateTime.GameTime % DateTime.Seconds(32) == 0 then
+		Media.DisplayMessage(Messages[4], "Mentat")
+	end
+end
+
+WorldLoaded = function()
+	player = Player.GetPlayer("Atreides")
+	harkonnen = Player.GetPlayer("Harkonnen")
+
+	if Map.Difficulty == "Normal" then
+		HarkonnenAttackWaves = 5
+		ToHarvest = 3000
+	elseif Map.Difficulty == "Hard" then
+		HarkonnenAttackWaves = 12
+		ToHarvest = 3500
+		Actor.Create("WormSpawner", true, { Owner = Player.GetPlayer("Creeps"), Location = WormWaypoint.Location })
+	end
+
+	InitObjectives()
+
+	Actor.Create(tostring(Map.Difficulty), true, { Owner = player })
+
+	Trigger.OnRemovedFromWorld(AtreidesConyard, function()
+		local refs = Map.ActorsInBox(Map.TopLeft, Map.BottomRight, function(actor)
+			return actor.Type == "refinery"
+		end)
+
+		if #refs == 0 then
+			harkonnen.MarkCompletedObjective(KillAtreides)
+		else
+			Trigger.OnAllRemovedFromWorld(refs, function()
+				harkonnen.MarkCompletedObjective(KillAtreides)
+			end)
+		end
+	end)
+
+	Media.DisplayMessage(Messages[1], "Mentat")
+
+	Trigger.AfterDelay(DateTime.Seconds(25), function()
+		Media.PlaySpeechNotification(player, "Reinforce")
+		Reinforcements.Reinforce(player, AtreidesReinforcements, AtreidesEntryPath)
+	end)
+
+	SendReinforcements()
+end
+
+SendReinforcements = function()
+	local units = HarkonnenReinforcements[Map.Difficulty]
+	local delay = Utils.RandomInteger(HarkonnenAttackDelay - DateTime.Seconds(2), HarkonnenAttackDelay)
+	HarkonnenAttackDelay = HarkonnenAttackDelay - (#units * 3 - 3 - HarkonnenAttackWaves) * DateTime.Seconds(1)
+	if HarkonnenAttackDelay < 0 then HarkonnenAttackDelay = 0 end
+
+	Trigger.AfterDelay(delay, function()
+		Reinforcements.Reinforce(harkonnen, Utils.Random(units), { Utils.Random(HarkonnenEntryWaypoints) }, 10, IdleHunt)
+
+		HarkonnenAttackWaves = HarkonnenAttackWaves - 1
+		if HarkonnenAttackWaves == 0 then
+			Trigger.AfterDelay(DateTime.Seconds(1), function() HarkonnenArrived = true end)
+		else
+			SendReinforcements()
+		end
+	end)
+end
+
+InitObjectives = function()
+	Trigger.OnObjectiveAdded(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "New " .. string.lower(p.GetObjectiveType(id)) .. " objective")
+	end)
+
+	KillAtreides = harkonnen.AddPrimaryObjective("Kill all Atreides units.")
+	GatherSpice = player.AddPrimaryObjective("Harvest " .. tostring(ToHarvest) .. " Solaris worth of Spice.")
+	KillHarkonnen = player.AddSecondaryObjective("Eliminate all Harkonnen units and reinforcements\n in the area.")
+
+	Trigger.OnObjectiveCompleted(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective completed")
+	end)
+	Trigger.OnObjectiveFailed(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective failed")
```

Comment:
"Objective completed/failed" are only those prefixes.
So full-stops are at the end of the messages, as they are at the end of the objective descriptions.